frontend/src/concepts/pipelines/content/createRun/RunPage.tsx

```diff
@@ -79,6 +83,15 @@ const RunPage: React.FC<RunPageProps> = ({
     [isSchedule, triggerType],
   );
 
+  const versionToUseData = React.useMemo<PipelineVersionToUse>(() => {
+    if (Object.values(PipelineVersionToUse).includes(contextVersionToUse)) {
+      return contextVersionToUse;
+    }
```

Comment:
Is this really needed? I think the `return` statement below should be able to cover all the situations. Could you give me an example to help me understand any scenarios I missed? 🤔

Reply:
This is to prioritize the prop that comes from the context, if provided, rather than trying to resolve the value through location or context versions. I figured it could be useful but at the moment there's no specific reason to have it. I'll remove it.